src/main/java/com/craftapi/bungeencpnotify/BungeeNCPNotify.java

```diff
@@ -66,29 +87,63 @@ public boolean isStaffMemberOnline() {
 	}
 
 	@Override
-	public void onLoad() {
+	public void onEnable() {
+
 		instance = this;
-	}
 
-	@Override
-	public void onEnable() {
 		// Config stuff
 		this.saveDefaultConfig();
 		configurationManager = new ConfigurationManager();
 
-		// Register NCP Hook and outgoing plugin channel if it's enabled in
-		// config
-		if (getConfiguration().getBoolean("general.enable-send")) {
+		// Other stuff
+		cooldownManager = new CooldownManager();
+		gson = new Gson();
+		SERVER_NAME = "";
+
+
+		// Fetch the server's name (only for reloads)
+		if (getServer().getPluginManager().getPlugin("LilyPad-Connect") != null) {
+			Connect connect = this.getServer().getServicesManager().getRegistration(Connect.class).getProvider();
+			SERVER_NAME = connect.getSettings().getUsername();
+		} else {
 			this.getServer().getMessenger().registerOutgoingPluginChannel(this, "BungeeCord");
-			NCPHookManager.addHook(CheckType.values(), new NotifyHook());
+			this.getServer().getMessenger().registerIncomingPluginChannel(this, "BungeeCord", new BungeeGetServer());
+			BungeeRequest.sendBungeeRequest("GetServer");
 		}
 
-		// Register incoming plugin channel if it's enabled in config
-		if (getConfiguration().getBoolean("general.enable-receive"))
-			this.getServer().getMessenger().registerIncomingPluginChannel(this, "BungeeCord", new IncomingMessageListener());
+		// If we are using redis then create the connection
+		if (getConfiguration().getBoolean("redis.enable")) {
+			if (getConfiguration().getString("redis.password") == null || getConfiguration().getString("redis.password").equals(""))
+				setJedisPool(new JedisPool(new JedisPoolConfig(), getConfiguration().getString("redis.ip"), getConfiguration().getInt("redis.port"), 0));
+			else
+				setJedisPool(new JedisPool(new JedisPoolConfig(), getConfiguration().getString("redis.ip"), getConfiguration().getInt("redis.port"), 0, getConfiguration().getString("redis.password")));
+		}
 
-		// Fetch the server's name (only for reloads)
-		BungeeRequest.sendBungeeRequest("GetServer");
+		// Register NCP Hook outgoing messages if it's enabled in config
+		if (getConfiguration().getBoolean("general.enable-send")) {
+			NCPHookManager.addHook(CheckType.values(), new NotifyHook());
+		}
+
+		// Register incoming message channel if it's enabled in config
+		if (getConfiguration().getBoolean("general.enable-receive")) {
+			if (getConfiguration().getBoolean("redis.enable")) {
+				new BukkitRunnable() {
+					@Override
+					public void run() {
+						setJedisListener(new IncomingRedisMessageListener());
+						Jedis jedis = getJedisPool().getResource();
+						try {
+							jedis.subscribe(getJedisListener(), "bncpnotify");
+						} catch (Exception e) {

```

Comment:
Don't catch general exceptions, catch specific exceptions and handle them individually.